src/network/connection_handler_task.cpp

```diff
@@ -14,17 +14,22 @@ ConnectionHandlerTask::ConnectionHandlerTask(const int task_id,
 }
 
 void ConnectionHandlerTask::Notify(int conn_fd, std::unique_ptr<ProtocolInterpreter> protocol_interpreter) {
-  client_fd_ = conn_fd;
-  protocol_interpreter_ = std::move(protocol_interpreter);
+  {
```

Comment:
Add a comment that explains why we need the latch in case some braze soul tries to remove it in the future.